ecosystem/typescript/sdk_v2/src/bcs/serializer.ts

```diff
@@ -12,8 +12,23 @@ import {
 } from "./consts";
 import { AnyNumber, Uint16, Uint32, Uint8 } from "../types";
 
-export interface Serializable {
-  serialize(serializer: Serializer): void;
+// This class is intended to be used as a base class for all serializable types.
+// It can be used to facilitate composable serialization of a complex type and
+// in general to serialize a type to its BCS representation.
+export abstract class Serializable {
+  abstract serialize(serializer: Serializer): void;
+
+  /**
+   * Serializes a `Serializable` value to its BCS representation.
+   * This function is the Typescript SDK equivalent of `bcs::to_bytes` in Move.
+   * @param value The Serializable value to serialize
```

Comment:
might be confusing saying the this function has a `value` param but irl it doesnt accept any argument, can we explain it better?

Reply:
Yeah I think that was a copy paste error 😅